### PR TITLE
More Compare Methods: cyan-magenta and blend

### PR DESCRIPTION
**Pull request by PaulHax**
Adds docs for `compare_images`, support for `image_mix` param, and another example cell in ThinPlateSpline notebook.

Depends on:
https://github.com/Kitware/itk-vtk-viewer/pull/668

TODO
- [x] Bump itk-vtk-viewer and bootstrap-ui pointers in viewer_config.py

### CODE REVIEW

File: itkwidgets/viewer.py [hunk 1]
```diff
@@ -467,9 +478,42 @@ def view(data=None, **kwargs):
...
+
+    method: string, default: None, possible values: 'cyan-magenta', 'blend', 'checkerboard'
+        checkerboard method picks pixels from the fixed and moving image to create a
+        checkerboard pattern.
+        cyan-magenta method puts the fixed image on component 0, moving image on component 1
+        and changes the color map for fixed image to cyan, moving image to magenta.
+        blend method puts the fixed image on component 0, moving image on component 1
+        and changes the color maps for both to grayscale.
+
+    imageMix: float, default: None
```
Comment (PaulHax) [inline itkwidgets/viewer.py:502]: snake case

File: itkwidgets/viewer.py [hunk 2]
```diff
@@ -467,9 +478,42 @@ def view(data=None, **kwargs):
...
+    
+    checkerboard: bool, default: None
+        Forces checkerboard pattern for cyan-magenta and blend methods.
+
+    pattern: Tuple[int, int, int], default: None
+        An array with the number of checkerboard boxes for each dimension.
+
+    swap_image_order: bool, default: None
+        Reverses which image is sampled for each checkerboard box.
+
```
Comment (PaulHax) [inline itkwidgets/viewer.py:515]: delete

File: itkwidgets/viewer.py [hunk 3]
```diff
@@ -467,9 +478,42 @@ def view(data=None, **kwargs):
 
     return viewer
 
-
 def compare_images(*args, **kwargs):
```
Comment (PaulHax) [inline itkwidgets/viewer.py:481]: Add the arguments of viewer.compare_images here

File: itkwidgets/viewer.py [hunk 4]
```diff
@@ -467,9 +478,42 @@ def view(data=None, **kwargs):
...
+    
+    Parameters
+    ----------
+    fixed_image: array_like, itk.Image, or vtk.vtkImageData
+        Static image the moving image is re-sampled to.
+
+    moving_image: array_like, itk.Image, or vtk.vtkImageData
+        Image is re-sampled to the fixed_image.
+
+    method: string, default: None, possible values: 'cyan-magenta', 'blend', 'checkerboard'
```
Comment (PaulHax) [inline itkwidgets/viewer.py:494]: document default derived args `image_mix` and `checkerboard`.

File: itkwidgets/viewer.py [hunk 5]
```diff
@@ -275,7 +275,7 @@ def set_image_volume_sample_distance(self, distance: float):
     def set_image_volume_scattering_blend(self, scattering_blend: float):
         self.queue_request('setImageVolumeScatteringBlend', scattering_blend)
 
-    def compare_images(self, fixed_image: Union[str, Image], moving_image: Union[str, Image], method: str = 'checkerboard', pattern: Tuple[int, int, int] = [4, 4, 4], swap_image_order: bool = False):
+    def compare_images(self, fixed_image: Union[str, Image], moving_image: Union[str, Image], method: str = None, image_mix: float = None, checkerboard: bool = None, pattern: Tuple[int, int, int] = None, swap_image_order: bool = None):
```
Comment (thewtex) [inline itkwidgets/viewer.py:278]: `pattern: Union[Tuple[int, int], Tuple[int, int, int]]`

File: itkwidgets/viewer.py [hunk 6]
```diff
@@ -467,10 +478,45 @@ def view(data=None, **kwargs):
 
     return viewer
 
+def compare_images(fixed_image: Union[str, Image], moving_image: Union[str, Image], method: str = None, image_mix: float = None, checkerboard: bool = None, pattern: Tuple[int, int, int] = None, swap_image_order: bool = None):
```
Comment (thewtex) [inline itkwidgets/viewer.py:481]: pattern type update

File: itkwidgets/viewer.py [hunk 7]
```diff
@@ -467,10 +478,45 @@ def view(data=None, **kwargs):
...
+        render by modifying the opacity transfer function. Value of 1 means max opacity for
+        moving image, 0 for fixed image. If value is None and the method is "blend" or "cyan-magenta",
+        the image_mix is set to 0.5.  If the method is "checkerboard", the image_mix is set to 0.
+    
+    checkerboard: bool, default: None
+        Forces the checkerboard mixing of fixed and moving images for the cyan-magenta and blend methods.
+        The rendered image has 2 components, each component reverses which image is sampled for each
+        checkerboard box.
+
+    pattern: Tuple[int, int, int], default: None
```
Comment (thewtex) [inline itkwidgets/viewer.py:512]: type update

File: itkwidgets/viewer.py [hunk 8]
```diff
@@ -467,10 +478,45 @@ def view(data=None, **kwargs):
...
+        The number of checkerboard boxes for each dimension.
+
+    swap_image_order: bool, default: None
+        Reverses which image is sampled for each checkerboard box.  This simply toggles
+        image_mix between 0 and 1.
+    """
     viewer = view()
-    viewer.compare_images(*args, **kwargs)
+    viewer.compare_images(fixed_image, moving_image, method, image_mix, checkerboard, pattern, swap_image_order)
```
Comment (thewtex) [inline itkwidgets/viewer.py:520]: keyword arguments to keyword arguments (more resilience to changes in argument order)

**Review by thewtex — COMMENTED** [review]
LGTM!

A few minor tweaks inline.

**Comment by PaulHax** [comment]
<p>I just updated the notebook so each of the 3 example cells show a different method.  Any other example ideas? </p>

---

 <a href='https://app.reviewnb.com/InsightSoftwareConsortium/itkwidgets/pull/642/discussion/#comment-1516552089'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by thewtex** [comment]
<p>Looks great!</p>

---

 <a href='https://app.reviewnb.com/InsightSoftwareConsortium/itkwidgets/pull/642/discussion/#comment-1516552089'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Pull request merged**